packages/table/components/Table.tsx

```diff
@@ -0,0 +1,142 @@
+    (
+      children: Array<React.ReactElement<IColumnProps>>,
+      width: number
+    ): number[] => {
+      const totalColumns: number = children.length;
+      let remainingWidth: number = width;
+      return children.map((child, currentIndex) => {
+        const calculatedWidth = child.props.width({
+          width,
+          totalColumns,
+          currentIndex,
+          remainingWidth
+        });
+        remainingWidth -= calculatedWidth;
+        return calculatedWidth;
+      });
+    }
+  );
+
+  constructor(props) {
+    super(props);
+
+    this.getCell = this.getCell.bind(this);
+    this.setRef = this.setRef.bind(this);
+    this.updateGrid = this.updateGrid.bind(this);
+  }
+
+  public getCell({ key, columnIndex, rowIndex, style }) {
+    const data = this.getData(this.props.data);
+
+    const column = React.Children.toArray(this.props.children)[
+      columnIndex
+    ] as React.ReactElement<IColumnProps>;
+
+    if (!column || column.type !== Column) {
+      return null;
+    }
+
+    let className = css({
+      borderBottom: `1px solid ${Colors().greyLight}`,
+      boxSizing: "border-box",
+      ...style
+    });
+
+    if (rowIndex === 0) {
+      className = css({
+        boxSizing: "border-box",
+        borderBottom: `1px solid ${Colors().black}`,
+        ...style
+      });
+      return <div {...{ className, key }}>{column.props.label}</div>;
+    }
+
+    return (
+      <div {...{ className, key }}>
+        {column.props.cellRenderer(data[rowIndex])}
+      </div>
+    );
+  }
+
+  public updateGrid() {
+    if (this.multiGridRef.recomputeGridSize != null) {
+      this.multiGridRef.recomputeGridSize();
+    }
+  }
+
+  public setRef(ref: any) {
+    this.multiGridRef = ref;
+  }
+
+  public render() {
+    const {
+      fixedColumnCount,
+      enableFixedColumnScroll,
+      enableFixedRowScroll
+    } = this.props;
+    const mergedData = this.getData(this.props.data);
+    const columnCount = React.Children.count(this.props.children);
+
+    return (
+      <AutoSizer onResize={this.updateGrid}>
+        {({ width, height }) => {
+          const columnSizes = this.getColumnSizes(
+            React.Children.toArray(this.props.children) as Array<
+              React.ReactElement<IColumnProps>
+            >,
+            width
+          );
+
+          const getColumnSize = ({ index }) => columnSizes[index];
+          return (
+            <MultiGrid
+              ref={this.setRef}
+              fixedColumnCount={fixedColumnCount}
+              fixedRowCount={1}
+              cellRenderer={this.getCell}
+              columnWidth={getColumnSize}
+              columnCount={columnCount}
+              enableFixedColumnScroll={enableFixedColumnScroll}
+              enableFixedRowScroll={enableFixedRowScroll}
+              height={height}
+              rowHeight={35}
+              rowCount={mergedData.length}
+              width={width}
+              hideTopRightGridScrollbar={true}
```

Comment:
I think you could also leave out the `={true}` part for these boolean properties if I remember JSX correctly

Reply:
that really defines which defaults are set right? I wanted to be over verbal here not if they change something our stuff breaks.